foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClient.java

```diff
@@ -126,13 +130,24 @@ public OpenAiClient withSystemPrompt(@Nonnull final String systemPrompt) {
    * @throws OpenAiClientException if the request fails
    */
   @Nonnull
-  public OpenAiChatCompletionOutput chatCompletion(@Nonnull final String prompt)
+  public CreateChatCompletionResponse chatCompletion(@Nonnull final String prompt)
       throws OpenAiClientException {
-    final OpenAiChatCompletionParameters parameters = new OpenAiChatCompletionParameters();
+    final CreateChatCompletionRequest parameters = new CreateChatCompletionRequest();
+
     if (systemPrompt != null) {
-      parameters.addMessages(new OpenAiChatSystemMessage().setContent(systemPrompt));
+      parameters.addMessagesItem(
+          new ChatCompletionRequestSystemMessage()
+              .role(ChatCompletionRequestSystemMessage.RoleEnum.SYSTEM)
+              .content(ChatCompletionRequestSystemMessageContent.create(systemPrompt)));
     }
-    parameters.addMessages(new OpenAiChatUserMessage().addText(prompt));
+    parameters
+        .addMessagesItem(
+            new ChatCompletionRequestUserMessage()
+                .role(ChatCompletionRequestUserMessage.RoleEnum.USER)
+                .content(ChatCompletionRequestUserMessageContent.create(prompt)))
+        .functions(null)
+        .tools(null)
+        .parallelToolCalls(null);
     return chatCompletion(parameters);
   }
```

Comment:
(Comment)

This is the expected diff for regular usage.
Without convenience API.